src/basic/terminal-util.c

```diff
@@ -972,42 +973,50 @@ int get_ctty(pid_t pid, dev_t *ret_devnr, char **ret) {
 
         r = device_path_make_canonical(S_IFCHR, devnr, &fn);
         if (r < 0) {
+                _cleanup_free_ char *pty = NULL;
```

Comment:
I don't think this can make any practical difference. If everybody hardcodes 136, 136 cannot be changed.
But if we do this, let's at least avoid the allocation:
```C
char pty[STRLEN("/dev/pts/") + DECIMAL_STR_MAX(dev_t)];
...
xsprintf(pty, ...);
```